src/python/pants/backend/codegen/tasks/protobuf_gen.py

```diff
@@ -103,55 +86,44 @@ def resolve_deps(self, deps_list, key):
                                   .format(message=e, key=key))
     return deps
 
+  def invalidate_for_files(self):
+    return [self.protobuf_binary]
+
   @property
   def javadeps(self):
     return self.resolve_deps(self.get_options().javadeps, 'javadeps')
```

Comment:
I just left it in because test_protobuf_gen already had working unit tests for it, and I didn't see a reason to break that.